dao/src/main/java/org/thingsboard/server/dao/timeseries/CassandraBaseTimeseriesDao.java

```diff
@@ -417,10 +381,34 @@ private ListenableFuture<List<Long>> getPartitionsFuture(TenantId tenantId, Read
         if (isFixedPartitioning()) { //no need to fetch partitions from DB
             return Futures.immediateFuture(FIXED_PARTITION);
         }
+        if (isPartitioningAlwaysExistInReading()) {
+            return Futures.immediateFuture(calculatePartitions(minPartition, maxPartition));
+        }
         TbResultSetFuture partitionsFuture = fetchPartitions(tenantId, entityId, query.getKey(), minPartition, maxPartition);
         return Futures.transformAsync(partitionsFuture, getPartitionsArrayFunction(), readResultsProcessingExecutor);
     }
 
+    List<Long> calculatePartitions(long minPartition, long maxPartition) {
+        if (minPartition == maxPartition) {
+            return Collections.singletonList(minPartition);
+        }
+        List<Long> partitions = new ArrayList<>();
+        partitions.add(minPartition);
+
+        long currentPartition = minPartition;
+        while (maxPartition > (currentPartition = calculateNextPartition(currentPartition))) {
+            partitions.add(currentPartition);
+        }
+
+        partitions.add(maxPartition);
+        return partitions;
+    }
+
+    private long calculateNextPartition(long ts) {
+        LocalDateTime time = LocalDateTime.ofInstant(Instant.ofEpochMilli(ts), ZoneOffset.UTC);
```

Comment:
add time as a parameter to avoid parsing the LocalDateTime on each call. We can reuse the LocalDateTime in the calculatePartitions